src/components/BooksQuery.tsx

```diff
@@ -29,29 +29,31 @@ const BooksQuery = () => {
       {isError && <div className="error">Error occured</div>}
       <div className="books">
         {response?.map((book) => (
-          <div key={book.id} className="books-list">
-            <div className="book-title">
-              <div className="id">
-                <strong>Title :</strong>
+          <Link key={book.id} state={book} to={`/book/${book.id}`}>
```

Comment:
Do not use `state` for doing this. What if a user shared a link to your website?

your_website/book/BOOK_ID

The data will not exist ❌